### ISSUE TITLE
Missing term in training AP+XP

### ISSUE THREAD

**Issue by kazewong** [jiadonglee/aspgap#1]
In the script `train_apxp.py`, the current training doesn't seem to do the prediction task with XP.

https://github.com/jiadonglee/xpformer/blob/5e30a9d416fe5389a41283d0ecc642c84bf55517/train_apxp.py#L33-L50

Instead of having the current line 41 and 42, the code should look something like

```
latent_embedding = encoder(XP)
AP = hallucinator(latent_embedding)
y1 = pretrained_decoder(AP)
y2 = decoder(latent_embedding)

loss = weight_1*loss(truth,y1) + weight_2*loss(truth,y2) 
```

Also, as a side note, why are you using a transformer for this now? I think an MLP should be sufficient and it should be easier to prove the architecture is working.

**Comment by jiadonglee** [comment]
In the latest push, I modified the codes into your suggestion, you can see it in line 41-48 of train_apxp.py, the overfitting issue still exists.

**Comment by jiadonglee** [comment]
And the new script is using MLP

**Comment by kazewong** [comment]
For the overfitting, do you have a plot showing the training and validation loss over training time? I also don't see where you checkpoint the model which has the lowest validation loss. That could be useful to look at.

Also, how many parameters are there in your model?

**Comment by jiadonglee** [comment]
<img width="600" alt="loss" src="https://user-images.githubusercontent.com/24669895/223448583-cfba9b28-fbca-439d-a01a-8ef876666852.png">
here is the loss function of train and validation
